cadc-storage-adapter/src/main/java/org/opencadc/inventory/storage/StorageClient.java

```
@@ -0,0 +1,195 @@
+        adapter.delete(storageID);
+    }
+
+    public Iterator<StorageLocation> iterator() throws TransientException {
+        return adapter.iterator();
+    }
+    
+    private void ioLoop(OutputStream out, InputStream in) {
+        // TODO: Write 2 threaded io loop--one thread reading
+        // and one thread writing.
+    }
+    
+    /**
+     * Load the configured storage adapter or the default one if none
+     * are found.
+     * 
+     * @return The storage adapter to be used by this client.
+     */
+    private StorageAdapter getStorageAdapter() {
+        Class c = null;
+        
+        // Try to load the adapter based on a classname in a system property
+        String cname = System.getProperty(StorageAdapter.class.getName());
+        if (cname != null) {
+            try {
+                c = Class.forName(cname);
+            } catch (Throwable t) {
+                String msg = "Failed to find storage adapter " + cname;
+                log.error(msg, t);
+                throw new IllegalStateException(msg, t);
+            }
+        }
+        
+        // Try to load the adapter based on the default name
+        cname = StorageAdapter.class.getName() + "Impl";
+        try {
+            c = Class.forName(cname);
+            try {
+                Constructor con = c.getConstructor();
+                Object o = con.newInstance();
+                StorageAdapter ret = (StorageAdapter) o;
+                log.debug("Loaded Storage adapter: " + cname);
+                return ret;
+            } catch (Throwable t) {
+                log.error("Failed to load storage adapter: " + cname, t);
+            }
+        } catch (Throwable t) {
+            String msg = "Failed to find storage adapter " + cname +
+                "  Using default implementation.";
+            log.debug(msg, t);
+        }
+        
+        StorageAdapter ret = new FileSystemStorageAdapter();
```

Should fail if not configured.